packages/react/src/components/provider.tsx

```diff
@@ -0,0 +1,44 @@
+import { useState, useMemo, useEffect, type ReactNode } from 'react';
+import { PenumbraProviderNotAvailableError, PenumbraSymbol } from '@penumbra-zone/client';
+import { penumbraContext } from '../hooks/penumbra';
+
+export const PenumbraProvider = ({
+  children,
+  providerOrigin,
+  requestApproval = false,
+  required = false,
+}: {
+  children?: ReactNode | ReactNode[];
```

Comment:
suggestion: leave only `children?: ReactNode`. It already includes `Iterable<ReactNode>` under the hood, so the array is not needed